php/ext/google/protobuf/message.c

```diff
@@ -240,29 +242,41 @@ static zval *Message_read_property(zval *obj, zval *member, int type,
  * The C extension version of checkInt32() doesn't actually check anything, so
  * we perform all checking and conversion in this function.
  */
-static void Message_write_property(zval *obj, zval *member, zval *val,
-                                   void **cache_slot) {
-  Message* intern = (Message*)Z_OBJ_P(obj);
+static PROTO_RETURN_VAL Message_write_property(
+    PROTO_VAL *obj, PROTO_STR *member, zval *val, void **cache_slot) {
+  Message* intern = PROTO_MSG_P(obj);
   const upb_fielddef *f = get_field(intern, member);
   upb_arena *arena = Arena_Get(&intern->arena);
   upb_msgval msgval;
 
-  if (!f) return;
+  if (!f) goto error;
 
   if (upb_fielddef_ismap(f)) {
     msgval.map_val = MapField_GetUpbMap(val, f, arena);
-    if (!msgval.map_val) return;
+    if (!msgval.map_val) goto error;
   } else if (upb_fielddef_isseq(f)) {
     msgval.array_val = RepeatedField_GetUpbArray(val, f, arena);
-    if (!msgval.array_val) return;
+    if (!msgval.array_val) goto error;
   } else {
     upb_fieldtype_t type = upb_fielddef_type(f);
     const Descriptor *subdesc = Descriptor_GetFromFieldDef(f);
     bool ok = Convert_PhpToUpb(val, &msgval, type, subdesc, arena);
-    if (!ok) return;
+    if (!ok) goto error;
   }
 
   upb_msg_set(intern->msg, f, msgval, arena);
+#if PHP_VERSION_ID < 80000
```

Comment:
I think this might be a PHP 7.4 change, not 8.0:
https://github.com/php/php-src/blob/PHP-7.4.0/UPGRADING.INTERNALS#L171-L173

Do we have any tests for 7.4?

Reply:
We have tests https://github.com/protocolbuffers/protobuf/blob/master/tests.sh#L665
This change just causes warning so far